### ISSUE TITLE
Base directive does not get parsed from turtle files.

### ISSUE THREAD

**Issue by BonaBeavis** [RDFLib/rdflib#2271]
The output is `None`, should it not be `http://example.org/`?

```
from rdflib import Graph

# create a Graph object
g = Graph()

# define an inline Turtle string with a base URI
turtle_str = """
@base <http://example.org/> .

<subject> <predicate> "literal" .
"""

# parse the Turtle string and set the base URI
g.parse(data=turtle_str, format="turtle")

# get the base URI of the Turtle string
base_uri = g.base

print(base_uri)
```

**Comment by aucampia** [comment]
Seems like a reasonable expectation, will look into it when I can.

**Comment by aucampia** [comment]
Actually, thinking about it now, I'm not sure this expectation makes sense. What if someone parses multiple turtle documents with different base directives? Should the graph's base change every time? The way things work now is, the @base directive is used in parsing, and the generated triples get added to the graph, there is very little other interaction, and this approach makes for a quite clean separation of concerns.

Would be happy to get more feedback from others.

**Comment by BonaBeavis** [comment]
You could do it like this, when the turtle file us parsed during creation of the graph with the constructor, the base uri is set, if it is parsed after the creation of the graph, the base uri is not set.

How can I access the base uri for setting the base manually?

**Comment by kouralex** [comment]
I agree with https://github.com/RDFLib/rdflib/issues/2271#issuecomment-1475286092. One can provide the base IRI (when serializing a graph) via the `base` parameter. https://github.com/RDFLib/rdflib/blob/ad56044cf3033f8d2a3c625d322f3d091d7c9322/rdflib/graph.py#L1286-L1324
